pkg/cloud/api/check.go

```diff
@@ -171,3 +171,44 @@ func checkSchema(t reflect.Type) error {
 
 	return nil
 }
+
+// CheckStructuralSubset checks if type From is the subset of To type. Type is a
+// subset of another if it contains fields with the same type and name. This
+// function is not symmetric. Path parameter is used for better error reporting.
+func CheckStructuralSubset(p Path, from, to reflect.Type) error {
```

Comment:
CheckStructuralSubset(from, to reflect.Type) calls

  checkStructuralSubsetImpl(Path{})

checkStructuralSubsetImpl(p Path, from, to)